src/components/utils/CodeFragment.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <div
+    ref="content"
+    class="code-preview"
+  >
+    <pre
+      v-for="row in lines"
+      :key="row.line"
+      class="code-preview__line"
+      :class="{'code-preview__line--current': isCurrentLine(row.line), [syntax]: true }"
+    ><span
+class="code-preview__line-num"
+           :data-line="row.line"
+    /><code>{{ row.content }}</code></pre>
+  </div>
+</template>
+
+<script>
+import hljs from 'highlight.js';
+
+/**
+ * This component is using to render some code fragment, for example in stack trace description
+ * It requires the 'lines' property as array of {line: number, content: string}
+ */
+export default {
+  name: 'CodeFragment',
+  props: {
+    /**
+     * Array of code fragment lines
+     * @type {{line: number, content: string}[]}
+     */
+    lines: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+
+    /**
+     * Array of line numbers that should be highlighted
+     * @type {number[]}
+     */
+    linesHighlighted: {
+      type: Array,
+      default: null
+    },
+
+    /**
+     * Code language
+     */
+    lang: {
+      type: String,
+      default: 'plaintext'
+    }
+  },
+  data() {
+    return {
+      /**
+       * Highlighting syntax.
+       * Can be overridden in some cases, {@link mounted}
+       */
+      syntax: null
+    };
+  },
+  computed: {
+    /**
+     * Concatenated code lines
+     * @return {string}
+     */
+    code() {
+      return this.lines.map(line => line.content).join('\n');
+    }
+  },
+  /**
+   * Vue mounted hook. Used to render highlighting
+   */
+  mounted() {
+    /**
+     * Set highlight.js syntax name based on catcher language
+     * @see https://github.com/highlightjs/highlight.js/tree/master/src/languages
+     */
+    this.syntax = this.lang;
+
+    /**
+     * Sometimes the JavaScript error can be triggered from inline scripts in HTML markup
+     * We need to highlight such code fragments as 'html' syntax instead of 'javascript'
+     */
+    if (this.syntax === 'javascript' && this.isHtmlScope()) {
+      this.syntax = 'html';
+    }
+
+    if (this.syntax !== 'plaintext') {
+      /**
+       * Timeout used to prevent Vue override 'hljs' class
+       */
+      setTimeout(() => {
+        /**
+         * We supposed to highlight each line separately because the code can be trimmed.
+         * This looks more suitable than incorrect highlighting of a whole block;
+         */
+        this.$refs.content.querySelectorAll('pre').forEach(el => {
+          hljs.highlightBlock(el);
```

Comment:
I did not quite understand the need to process each line separately

Reply:
See comment above the code.

Highlighting of each line separatelly:
<img width="1076" alt="image" src="https://user-images.githubusercontent.com/3684889/70854498-d17d4c00-1ecd-11ea-9a5c-8252d0025aa0.png">

Highlighting of a whole block
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/3684889/70854519-04bfdb00-1ece-11ea-9eb1-31d48c06534a.png">